_posts/2022-08-15-changelog-176.markdown

```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #176"
+date: 2022-08-15 13:00:00 +0300
+---
+
+
+## Performance Improvements
+
+* [#9074] Slightly optimize name resolution
+
+* [#9027] Slightly speed up highlighting and appearance of inlay hints
+
+## Fixes
+
+* [#9169] Fix possible stack overflow introduced in the previous release of the plugin
+
+* [#9083] Resolve enum variants in match patterns properly
+
+* [#9106] Resolve reexported macros properly
+
+* [#9047] Fix various bugs in ["Inline function"](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#extractmethod-refactoring) refactoring related to the substitution of parameters and function bodies and name conflicts
+
+* [#9155] Do not suggest unexpected or duplicate completion variants in `Cargo.toml`
+
+* [#9152] Do not suggest macros when completing module names
+
+* [#9150] Fix handling of `self` parameters in [`Change signature`](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#change-sign) refactoring (by [@Kobzol])
+
+* [#9117] Fix [`Rename`](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#rename-refactoring) refactoring when invoked on a constant used in a [shorthand init](https://doc.rust-lang.org/reference/expressions/struct-expr.html#struct-field-init-shorthand)
+
+* [#9146] Disable `Extract struct fields` and `Generate getter/setter` actions when no fields are selected (by [@Kobzol])
```

Comment:
@Undin Looks like we don't have documentation for these actions, should we open a ticket for that?

Reply:
We have paragraph about `Generate getter/setter` [here](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-code-generation.html#generate-menu)

`Extract struct fields` is indeed not documented. CC @ksenia-starodubtseva